prusti-viper/src/encoder/procedure_encoder.rs

```diff
@@ -4805,7 +4808,7 @@ impl<'p, 'v: 'p, 'tcx: 'v> ProcedureEncoder<'p, 'v, 'tcx> {
             }
 
             &mir::AggregateKind::Closure(def_id, _substs) => {
-                assert!(self.encoder.is_spec_closure(def_id), "closure: {:?}", def_id);
+                //assert!(self.encoder.is_spec_closure(def_id), "closure: {:?}", def_id);
                 // Specification only. Just ignore in the encoding.
                 // FIXME: Filtering of specification blocks is broken, so we need to handle this here.
                 Vec::new()
```

Comment:
Shouldn't this be `unimplemented!()`?

Reply:
No. Arguably, maybe there needs to be something else there, but it's not "unimplemented", it's in the process of being implemented. If I put `unimplemented!()` there, I'll get a panic every time I run Prusti on some code that has a non-spec-only closure in it, which is precisely the feature I'm trying to implement.

Or did you mean I should _temporarily_ put `unimplemented!()` there for this pull request and remove it on my fork again immediately afterwards?

Reply:
Ideally, for this PR it should be something like this:
```suggestion
              if self.encoder.is_spec_closure(def_id) {
                // Specification only. Just ignore in the encoding.
                // FIXME: Filtering of specification blocks is broken, so we need to handle this here.
                Vec::new()
                } else {
                  unimplemented!();
                }
```